src/groupederror.jl

```diff
@@ -0,0 +1,137 @@
+type GroupedError
+    x
+    y
+    err
+    group
+    xaxis
+end
+
+get_axis(column::PooledDataArray) = sort!(unique(column))
+get_axis(column::AbstractArray) = linspace(minimum(column),maximum(column),200)
+
+# f is the function used to analyze dataset: define it as nan when it is not defined,
+# the input is: dataframe used, points chosen on the x axis, x (and maybe y) column labels
+# the output is the y value for the given xvalues
+
+function get_mean_sem(trend,variation, f, df, population, args...; xaxis = false, kwargs...)
+    # define points on x axis
+    if xaxis == false
+        xvalues = get_axis(df[args[1]])
+    else
+        xvalues = xaxis
+    end
+
+    if population == []
+        mean_across_pop = f(df,xvalues, args...; kwargs...)
+        sem_across_pop = zeros(length(xvalues));
+        valid = ~isnan(mean_across_pop)
+    else
+        # get mean value and sem of function of interest
+        splitdata = groupby(df, population)
+        v = Array(Float64, length(xvalues), length(splitdata));
+        for i in 1:length(splitdata)
+            v[:,i] = f(splitdata[i],xvalues, args...; kwargs...)
+        end
+        mean_across_pop = Array(Float64, length(xvalues));
+        sem_across_pop = Array(Float64, length(xvalues));
+        for j in 1:length(xvalues)
+            notnan = !isnan(v[j,:])
+            mean_across_pop[j] = trend(v[j,notnan])
+            sem_across_pop[j] = variation(v[j,notnan])
+        end
```

Comment:
Actually with real data these plots tend to have huge error bars towards the end, which is somewhat annoying, but it could also be that one has a lousy subject with a weird range and you don't want that to prevent you from looking at most of your data. As of now I require at least 2 subject to be not NaN to plot the mean and sem but that could be another keyword (in terms of number of subjects or fraction of subjects required to consider it a "plottable" point). In principle one could even think to show with some vertical lines or sth else where is the majority of the data (say from 0.05 to 0.95 quantile).